### PR TITLE
Update To Reset Behavior, Change SoftReset to Initialize

### PR DESCRIPTION
**Pull request by slacrherbst**
This PR changes the behavior of the Rogue reset commands and how they are used.

SoftReset now becomes Initialize

The three reset functions, HardReset, CountReset and Initialize are no longer recursed from the top level. Instead the default function in a device will recurse on the child devices. If the user re-implemented the reset function, they must call the super() reset function in order to properly recurse if desired. 

A new variable in root, InitAfterConfig, defines if the Initialize() method should be called after a yaml file is loaded, or after setYaml() is called directly. 

This PR also sets all blocks to stale as a result of HardReset and at startup..

ReadConfig is changed to LoadConfig, WriteConfig is changed to SaveConfig and WriteState is changed to SaveState.

### CODE REVIEW

File: python/pyrogue/_Block.py [hunk 1]
```diff
@@ -330,6 +330,9 @@ def __init__(self, *, offset, size, variables):
         # Set exclusive flag
         self._overlapEn = all (excMask[i] == 0 for i in range(size))
 
+        # Force block to be stale at startup
+        self._forceStale()
```
Comment (slacrherbst) [inline python/pyrogue/_Block.py]: Please not this change.

File: .travis.yml [hunk 2]
```diff
@@ -172,7 +172,8 @@ jobs:
 
       script:
         # Build conda package
-        - conda build -q conda-recipe --output-folder bld-dir -c defaults -c conda-forge -c paulscherrerinstitute
+        - conda build --debug conda-recipe --output-folder bld-dir -c defaults -c conda-forge -c paulscherrerinstitute
```
Comment (ruck314) [inline .travis.yml]: Is the `--debug` flag the new default?  Or was this an accidental comment?

Comment (slacrherbst) [reply .travis.yml]: Sneaking in a debug flag to figure out why the mac conda build is failing.

**Comment by bengineerd** [comment]
This PR should also change readConfig and writeConfig to loadConfig and saveConfig.

**Comment by slacrherbst** [comment]
Ok. I changed those functions. They were already named that in the GUI.

**Pull request merged**